### PR TITLE
[#1769] Implement more Checkstyle checks 

### LINKED ISSUE #1769: Implement more Checkstyle checks
**What feature(s) would you like to see in RepoSense?**

While working on https://github.com/se-edu/addressbook-level3/pull/133, I found additional checks that could be useful for us:
* [JavadocTagContinuationIndentation](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocTagContinuationIndentation) - Check for Google Java Style Guide requirement of [4 whitespaces for continuation lines in block tags](https://google.github.io/styleguide/javaguide.html#s7.1.3-javadoc-block-tags)
* [DefaultComesLast](https://checkstyle.sourceforge.io/config_coding.html#DefaultComesLast) - Ensures that all cases in `switch` statement are before `default`
* [NoWhitespaceBeforeCaseDefaultColon](https://checkstyle.sourceforge.io/config_whitespace.html#NoWhitespaceBeforeCaseDefaultColon) - E.g. `default :` would not be allowed

**If possible, describe the solution**

Before implementing any new checks, we should look through [other available checks](https://checkstyle.sourceforge.io/checks.html) and see which checks are suitable. By having as many checks as possible in a single PR, this will allow us to avoid revisiting this issue with multiple PRs in the future.

### PR DESCRIPTION
**Pull request by germainelee02**
Fixes #1769 

```
Currently, RepoSense does not have Checkstyle for 
- JavadocTagContinuationIndentation
- DefaultComesLast
- NoWhitespaceBeforeCaseDefaultColon.

Adding these checks can make it easier for future contributors 
to adhere to Reposense's coding style.

Let's add more Checkstyle checks to ensure easier standardisation
of code style.
```

- [JavadocTagContinuationIndentation](https://checkstyle.sourceforge.io/checks/javadoc/javadoctagcontinuationindentation.html#JavadocTagContinuationIndentation)
- [DefaultComesLast](https://checkstyle.sourceforge.io/config_coding.html#DefaultComesLast)
- [NoWhitespaceBeforeCaseDefaultColon](https://checkstyle.sourceforge.io/checks/whitespace/nowhitespacebeforecasedefaultcolon.html).

### CODE REVIEW

**Review by yhtMinceraft1010X — CHANGES_REQUESTED** [review]
Great work on identifying new checks.

For easier referencing by future developers, we group the checks into different sections below.

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -37,7 +37,6 @@
 
   <!-- All Java AST specific tests live under TreeWalker module. -->
   <module name="TreeWalker">
-
```
Comment (yhtMinceraft1010X) [inline config/checkstyle/checkstyle.xml:40]: Missing line. This may sound nitpicky, but the idea is that unnecessary changes should not be included in a PR.

So the only 3 changes should be green blocks corresponding to the new checks.

Comment (yhtMinceraft1010X) [reply config/checkstyle/checkstyle.xml:40]: Other than this, everything else looks good

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Review by ckcherry23 — APPROVED** [review]
LGTM!

**Comment by chan-j-d** [comment]
@germainelee02 Apologies as this is a little late but do note place the commit message in a code block as ckcherry has done for you. The links to the relevant documentation can be placed after the commit message together with any other relevant information for the PR.
